### PR TITLE
Solution properly working

### PR DESCRIPTION
**Pull request by dorotan**

### CODE REVIEW

**Comment by dorotan** [comment]
Solution properly working

**Comment by santiagobasulto** [comment]
This PR will be reviewed by: @naveeda11

File: dealership/contracts.py [hunk 1]
```diff
@@ -1,12 +1,68 @@
...
     def __init__(self, vehicle, customer, monthly_payments):
-        pass
+        self.vehicle = vehicle
+        self.customer = customer
+        self.monthly_payments = monthly_payments
+
+    def total_value(self):
+        i_rate = None 
+        
+        if self.vehicle.__class__.__name__ == "Car":
```
Comment (naveeda11) [inline dealership/contracts.py:22]: To determine the type of an object, you could use type() or isinstance. 
Example 
if isinstance(self.vehicle, Car):

File: dealership/contracts.py [hunk 2]
```diff
@@ -1,12 +1,68 @@
...
+            i_rate = 1.07
+        elif self.vehicle.__class__.__name__ == "Motorcycle":
+            i_rate = 1.03
+        else:
+            i_rate = 1.11
+        
+        t_price = self.vehicle.sale_price() + (i_rate * self.monthly_payments * self.vehicle.sale_price() / 100)
+            
+        if self.customer.__class__.__name__ == "Employee":
```
Comment (naveeda11) [inline dealership/contracts.py:31]: minor suggestion to use .9 as a variable labelled "employee_discount" so it is clear for any newcomer that the reason to multiply by .9 is to apply a discount

File: dealership/contracts.py [hunk 3]
```diff
@@ -1,12 +1,68 @@
...
+            
+        t_price = (self.vehicle.sale_price()) + (self.vehicle.sale_price() * lease_multiplier) / (self.length_in_months)
+        
+        if self.customer.__class__.__name__ == "Customer":
+            return t_price
+        else:
+            return t_price * 0.9
+
+
+    def monthly_value(self):
```
Comment (naveeda11) [inline dealership/contracts.py:65]: 1. Total_value() can be a static method @staticmethod
2. If total_value() is NOT a static method , then you don't need to prefix with LeaseContract. Instead you can prefix with self
I think this question can be asked in class. - I am not sure

**Pull request closed without merging**